src/main/resources/db/migration/V1__create_users_table.sql

```diff
@@ -0,0 +1,41 @@
+-- Create users table
+CREATE TABLE users (
+    id BIGSERIAL PRIMARY KEY,
+    email VARCHAR(255) NOT NULL UNIQUE,
+    password VARCHAR(255),
+    name VARCHAR(100) NOT NULL,
+    provider_type VARCHAR(10) NOT NULL,
+    role VARCHAR(255) NOT NULL DEFAULT 'ROLE_USER',
+    created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    updated_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP
+);
+
+-- Create indexes for better query performance
+CREATE INDEX idx_users_email ON users(email);
+CREATE INDEX idx_users_provider_type ON users(provider_type);
```

Comment:
```suggestion
CREATE UNIQUE INDEX uq_user_email_provider
ON users (email, provider_type);
```
unique key 제약 조건을 걸면 중복제거 + index가 가능할 것 같아 제안드립니다.